learning_resources/views.py

```diff
@@ -376,6 +379,129 @@ class ResourceListItemsViewSet(NestedViewSetMixin, viewsets.ReadOnlyModelViewSet
     ordering = ["position", "-child__last_modified"]
 
 
+@extend_schema(
+    parameters=[
+        OpenApiParameter(
+            name="id",
+            type=OpenApiTypes.INT,
+            location=OpenApiParameter.PATH,
+            description="id of the learning resource",
+        ),
+    ],
+)
+class LearningResourceListRelationshipViewSet(viewsets.GenericViewSet):
+    """
+    Viewset for managing relationships between Learning Resources
+    and User Lists / Learning Paths
+    """
+
+    permission_classes = (AnonymousAccessReadonlyPermission,)
+    filter_backends = [MultipleOptionsFilterBackend]
+    queryset = LearningResourceRelationship.objects.select_related("parent", "child")
+    http_method_names = ["patch"]
+
+    def get_serializer_class(self):
+        if self.action == "userlists":
+            return UserListRelationshipSerializer
+        elif self.action == "learning_paths":
+            return LearningResourceRelationshipSerializer
+        return super().get_serializer_class()
+
+    @extend_schema(
+        summary="Set User List Relationships",
+        description="Set User List Relationships on a given Learning Resource.",
+        parameters=[
+            OpenApiParameter(
+                name="userlist_id",
+                type=OpenApiTypes.INT,
+                many=True,
+                location=OpenApiParameter.QUERY,
+                description="id of the parent user list",
+            ),
+        ],
+        responses={200: UserListRelationshipSerializer(many=True)},
+    )
+    @action(detail=True, methods=["patch"], name="Set User List Relationships")
+    def userlists(self, request, *args, **kwargs):  # noqa: ARG002
+        """
+        Set User List relationships for a given Learning Resource
+        """
+        learning_resource_id = kwargs.get("pk")
+        user_list_ids = request.query_params.getlist("userlist_id")
```

Comment:
**Suggestion:**  Throw a 403 here if one of the userlists does not belong to user
Something like
```python
if (UserList.objects.filter(pk__in=user_list_ids).exclude(author=request.user).exists()):
    msg = f"You do not have permission to edit all of {user_list_ids}"
    raise PermissionDenied(msg)
```
and maybe add a test for the 403 error.

I think this is worthwhile and doesn't seem too complicated